src/OpenStudy/study_openinterface.jl

```diff
@@ -260,47 +435,139 @@ function get_parms(
     ignore::Bool = false,
     default::T = _default_value(T),
 ) where T
-    raw = _raw(data)
+
+    attr_struct = get_attribute_struct(data, col, name)
+    if check_type
+        _check_type(attr_struct, T, col, name)
+    end
+    if check_parm
+        _check_parm(attr_struct, col, name)
+    end
 
     n = max_elements(data, col)
-    if n == 0
-        return T[]
+    out = T[]
+    sizehint!(out, n)
+    for i in 1:n
+        push!(out, get_parm(data, col, name, i, T; default = default))
     end
```

Comment:
I did'nt understand this here, why not allocate the full vector with every entry and then make an attribution with `get_parm`?

Reply:
eve if it is a vector of vectors we could allocate the outer Vector structure